### PR TITLE
feat: feedback component submits to feedback table + tidy ups

### PR DESCRIPTION
**Pull request by jamdelion**
## In this PR:

Improvements from comments on [previous PR](https://github.com/theopensystemslab/planx-new/pull/3866):
- Fixed `more information` link not appearing as expect on public component
- Swapped `RichTextInput` for `TextInput` component in public view
- Restyled disclaimer using shared component
- Renamed `placeholders` file given will be used in templates, and made the object typeable.

Also:
- Fixed styling of face boxes uppercase text.
- Hook into existing feedback API (added 'feedback score' column to table)
- Also add `_feedback` to breadcrumbs on submit

## How to test 🧪 

- Turn on the feature flag with `window.featureFlags.toggle("FEEDBACK_COMPONENT").
- Try to add a Feedback component to a flow, publish it etc!

🎟️ Trello ticket: https://trello.com/c/jEOLoubW/3034-can-we-introduce-in-house-end-of-service-feedback

## Todo:

- [ ] e2e testing
- [ ]  Add feedback score to feedback log table, and add new feedback type (currently renders as default 'inaccuracy')
- [ ] validation and error handling, with testing
- [ ] use the `_feedback` key in metabase (add to `allow_list`s)
- [ ] make sure component works on 'back' - `previouslySubmittedProps`

### CODE REVIEW

File: editor.planx.uk/src/@planx/components/Feedback/Public/Public.tsx [hunk 1]
```diff
@@ -21,14 +26,39 @@ import { Feedback, FormProps } from "../model";
...
+    const metadata = await getInternalFeedbackMetadata();
+    const feedback = {
+      userComment: values.feedback,
+      feedbackScore: values.feedbackScore,
+    };
+    const data = {
+      ...metadata,
+      ...feedback,
+      feedbackType: "component" as FeedbackView,
+    };
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Feedback/Public/Public.tsx:42]:
```suggestion
    const metadata = await getInternalFeedbackMetadata();
    const data = {
      ...metadata,
      ...values,
    };
```

nit: We gain a small simplification and slightly lessen the mental load here by not having to convert `feedback` to `userComment`. I think we could use `userComment` as the field name to match the db column 👍 

Likewise, there's nothing stopping us adding the `feedbackType` field to the form and then not just presenting it to the user (not sure if this is actually a helpful step though!)

File: editor.planx.uk/src/@planx/components/Feedback/Public/Public.tsx [hunk 2]
```diff
@@ -21,14 +26,39 @@ import { Feedback, FormProps } from "../model";
...
+    const submitFeedbackResult = await insertFeedbackMutation(data).catch(
+      (err) => {
+        console.error(err);
+      },
+    );
+    props.handleSubmit?.(makeData(props, values, "_feedback"));
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Feedback/Public/Public.tsx:48]: I'd recommend making this `_feedback` key a bit more visible / discoverable

```ts
const PASSPORT_FEEDBACK_KEY = "_feedback" as const;
...
props.handleSubmit?.(makeData(props, values, PASSPORT_FEEDBACK_KEY));

```

As noted in the PR description, we'll want to re-use this value for the `ALLOW_LIST` for analytics.

File: editor.planx.uk/src/@planx/components/Feedback/Public/Public.tsx [hunk 3]
```diff
@@ -21,14 +26,39 @@ import { Feedback, FormProps } from "../model";
...
+    if (!submitFeedbackResult) {
+      return;
+    }
+
+    resetForm({ values });
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Feedback/Public/Public.tsx:53]: I don't think we need to worry about resetting the form - calling `props.handleSubmit()` will trigger the next card to be rendered when a breadcrumb is added.

**Review by DafyddLlyr — CHANGES_REQUESTED** [review]
Great stuff - working as expected 👍 

Code is clear and well structured, easy to review and follow along.

One thing for a follow up PR is to make sure that the component works on "back" - have a look at how the `previouslySubmittedData` prop is used in other components.

A few comments below - let me know if you want to talk anything through.

File: editor.planx.uk/src/@planx/components/Feedback/Public/Public.tsx (hunk 1 above)
Comment (jamdelion) [reply editor.planx.uk/src/@planx/components/Feedback/Public/Public.tsx:41]: Great suggestion! I think I would find having the feedbackType in the form more confusing to read in the code so have left that out though.

File: editor.planx.uk/src/@planx/components/Feedback/components/FaceBox.tsx [hunk 4]
```diff
@@ -42,7 +42,17 @@ export const FaceBox = ({
           })}
         >
           <img src={icon} width={50} alt={altText} />
-          <Typography variant="body2" pt={0.5}>
+          <Typography
```
[the remark this answers is not in the document]
Comment (jamdelion) [reply editor.planx.uk/src/@planx/components/Feedback/components/FaceBox.tsx:51]: I always forget that `Capitalize` exists! 💡 Good shout :)

File: editor.planx.uk/src/lib/feedback.ts [hunk 5]
```diff
@@ -55,7 +57,8 @@ export async function insertFeedbackMutation(data: {
   userData?: UserData;
   userContext?: string;
   userComment: string;
-  feedbackType: string;
+  feedbackScore?: string;
```
[the remark this answers is not in the document]
Comment (jamdelion) [reply editor.planx.uk/src/lib/feedback.ts:60]: Good shout, hadn't thought through this properly :) Have changed to an int with a check constraint

File: editor.planx.uk/src/@planx/components/Feedback/Editor/Editor.tsx [hunk 6]
```diff
@@ -89,8 +85,8 @@ export const FeedbackEditor = (props: FeedbackEditorProps) => {
               <InputLabel label="Disclaimer text" htmlFor="disclaimer">
                 <RichTextInput
                   name="disclaimer"
-                  value={formik.values.disclaimer || disclaimerPlaceholder}
-                  placeholder={disclaimerPlaceholder}
+                  value={formik.values.disclaimer || defaultContent.disclaimer}
+                  placeholder={defaultContent.disclaimer}
```
[the remark this answers is not in the document]
Comment (jamdelion) [reply editor.planx.uk/src/@planx/components/Feedback/Editor/Editor.tsx]: I've solved this by deleting the placeholder 😅

File: hasura.planx.uk/migrations/1730906349434_alter_table_public_feedback_add_column_feedback_score/up.sql [hunk 7]
```diff
@@ -0,0 +1,2 @@
+alter table "public"."feedback" add column "feedback_score" text
```
[the remark this answers is not in the document]
Comment (jamdelion) [reply hasura.planx.uk/migrations/1730906349434_alter_table_public_feedback_add_column_feedback_score/up.sql:1]: One thing I'm not sure of - can I delete some of these migrations to make the PR cleaner? Or is my back and forth on this immortalised forever more? 😅

[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply hasura.planx.uk/migrations/1730906349434_alter_table_public_feedback_add_column_feedback_score/up.sql:1]: Yes! Can manually combine migrations into a single `up.sql` (note you'll want to destroy Vultr before pushing commit so it rebuilds from scratch to test) or there's a [Hasura CLI squash command](https://hasura.io/docs/2.0/hasura-cli/commands/hasura_migrate_squash/) (I find this command finicky and usually go the manual route tbh!)

**Review by jessicamcinchak — APPROVED** [review]
Changes & previous feedback all look good here - happy to approve to not block progressing !

**Pull request merged**